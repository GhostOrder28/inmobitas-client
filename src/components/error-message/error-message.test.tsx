import ErrorMessage from './error-message.component';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'

describe('ErrorMessage', () => {
  it('should render with message', () => {
    const wrapper = render(<ErrorMessage fieldErrorMsg={'this is an error'} />)
    expect(wrapper).toMatchSnapshot();
  })

  it('should render nothing', () => {
    render(<ErrorMessage fieldErrorMsg={undefined} />);
    const errorMessageContainer = screen.queryByTestId('error-message-container');
    expect(errorMessageContainer).not.toBeInTheDocument();
  })
})
