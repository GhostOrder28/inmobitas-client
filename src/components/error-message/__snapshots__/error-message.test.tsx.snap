// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render with message 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      data-evergreen-toaster-container=""
    >
      <span
        class="css-1sugtjn"
      />
    </div>
    <div>
      <div
        class="ub-mt_-3rem ub-dspl_flex ub-algn-itms_center ub-box-szg_border-box"
        data-testid="error-message-container"
      >
        <svg
          class="css-tn1wco ub-mr_5px ub-w_13px ub-h_13px ub-box-szg_border-box"
          data-icon="ban-circle"
          viewBox="0 0 16 16"
        >
          <path
            d="M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm3 9H5c-.55 0-1-.45-1-1s.45-1 1-1h6c.55 0 1 .45 1 1s-.45 1-1 1z"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="ub-fnt-sze_-75rem ub-f-wght_400 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-fnt-fam_b77syt ub-color_474d66 ub-w_100prcnt ub-box-szg_border-box"
        >
          this is an error
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ub-mt_-3rem ub-dspl_flex ub-algn-itms_center ub-box-szg_border-box"
      data-testid="error-message-container"
    >
      <svg
        class="css-tn1wco ub-mr_5px ub-w_13px ub-h_13px ub-box-szg_border-box"
        data-icon="ban-circle"
        viewBox="0 0 16 16"
      >
        <path
          d="M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm3 9H5c-.55 0-1-.45-1-1s.45-1 1-1h6c.55 0 1 .45 1 1s-.45 1-1 1z"
          fill-rule="evenodd"
        />
      </svg>
      <span
        class="ub-fnt-sze_-75rem ub-f-wght_400 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-fnt-fam_b77syt ub-color_474d66 ub-w_100prcnt ub-box-szg_border-box"
      >
        this is an error
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
